### PR TITLE
make module_to_export_counts_folder

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] make module_to_export_counts_folder. It will put together AE and AS data and modify the sample annotation so that the newly created database is much easier. The produced sample annotation can now be used as input for the pipeline without further modification.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/drop_config_runAE.nf [hunk 1]
```diff
@@ -84,7 +84,9 @@ process DROP_CONFIG_RUN_AE {
     touch gene_name_mapping_.tsv
     mkdir output
     if [[ $skip_export_counts_drop == false ]]; then
-        mkdir exported_counts
+        mkdir exported_counts_ae
```
Comment (Lucpen) [inline modules/local/drop_config_runAE.nf:87]:
```suggestion
        mkdir -p exported_counts_ae
```

File: modules/local/drop_config_runAS.nf [hunk 2]
```diff
@@ -81,7 +81,13 @@ process DROP_CONFIG_RUN_AS {
     touch gene_name_mapping_.tsv
     mkdir output
     if [[ $skip_export_counts_drop == false ]]; then
-        mkdir exported_counts
+        mkdir exported_counts_as
```
Comment (Lucpen) [inline modules/local/drop_config_runAS.nf:84]:
```suggestion
        mkdir -p exported_counts_as
```

File: bin/drop_sample_annot_exported_counts.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+from pandas import read_csv, DataFrame
```
Comment (peterpru) [inline bin/drop_sample_annot_exported_counts.py:5]:
```suggestion
from pandas import read_csv, DataFrame
from pathlib import Path
```

File: bin/drop_sample_annot_exported_counts.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+SCRIPT_VERSION = "v1.0"
```
Comment (peterpru) [inline bin/drop_sample_annot_exported_counts.py:7]:
```suggestion
SCRIPT_VERSION = "v1.0"
GENE_COUNTS_PATH = Path("exported_counts/geneCounts.tsv.gz")
SPLICE_COUNTS_DIR = Path("exported_counts")
OUTPUT_FILE = Path("exported_counts/sampleAnnotation.tsv")
```

File: bin/drop_sample_annot_exported_counts.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    that can be used as input for Tomte
+    """
+    df_samples: DataFrame = read_csv(sample_annot, sep="\t")
+    df_samples["RNA_BAM_FILE"] = "NA"
+    df_samples["GENE_ANNOTATION"] = df_samples["GENE_ANNOTATION"].fillna(gtf)
+    df_samples = modify_gene_counts_df(
+        df=df_samples,
+        col_name="GENE_COUNTS_FILE",
+        run=ae_run,
+        value_in="exported_counts/geneCounts.tsv.gz",
```
Comment (peterpru) [inline bin/drop_sample_annot_exported_counts.py:34]:
```suggestion
        value_in=str(GENE_COUNTS_PATH),
```

File: bin/drop_sample_annot_exported_counts.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    )
+    df_samples = modify_gene_counts_df(
+        df=df_samples,
+        col_name="SPLICE_COUNTS_DIR",
+        run=as_run,
+        value_in="exported_counts",
```
Comment (peterpru) [inline bin/drop_sample_annot_exported_counts.py:40]:
```suggestion
        value_in=str(SPLICE_COUNTS_DIR),
```

File: bin/drop_sample_annot_exported_counts.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    )
+    df_samples.to_csv("exported_counts/sampleAnnotation.tsv", index=False, sep="\t")
```
Comment (peterpru) [inline bin/drop_sample_annot_exported_counts.py:42]:
```suggestion
    df_samples.to_csv(OUTPUT_FILE, index=False, sep="\t")
```

**Review by peterpru — APPROVED** [review]
I think it looks good! The suggestions I made in the python script is to change the file paths to constants, which makes them easier to manage and reuse.

File: modules/local/drop_put_together_exported_couts.nf [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+    $baseDir/bin/drop_sample_annot_exported_counts.py \\
+        --sample_annot "sample_annotation.tsv" \\
+        --ae_run $ae_run \\
+        --as_run $as_run \\
+        --gtf $gtf_no_extension
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        drop_sample_annotation_exported_counts: \$(\$baseDir/bin/drop_sample_annotation_exported_counts.py --version )
```
Comment (fellen31) [inline modules/local/drop_put_together_exported_couts.nf:50]:
```suggestion
        drop_sample_annotation_exported_counts: \$(drop_sample_annotation_exported_counts.py --version)
```

Same here

File: modules/local/drop_put_together_exported_couts.nf [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+    """
+
+    stub:
+    """
+    mkdir -p exported_counts
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        drop_sample_annotation_exported_counts: \$(\$baseDir/bin/drop_sample_annotation_exported_counts.py --version )
```
Comment (fellen31) [inline modules/local/drop_put_together_exported_couts.nf:61]:
```suggestion
        drop_sample_annotation_exported_counts: \$(drop_sample_annotation_exported_counts.py --version)
```

File: modules/local/drop_put_together_exported_couts.nf [hunk 10]
```diff
@@ -0,0 +1,64 @@
+process DROP_PUT_TOGETHER_EXPORTED_COUNTS {
+    tag "DROP_put_together_exported_couts"
```
Comment (fellen31) [inline modules/local/drop_put_together_exported_couts.nf:2]:
```suggestion
    tag "DROP_put_together_exported_counts"
```

**Review by fellen31 — COMMENTED** [review]
Tiny comments :)

**Review by fellen31 — APPROVED** [review]
🚀

**Pull request merged**